tiledb/sm/filesystem/posix.h

```diff
@@ -273,9 +273,9 @@ class Posix {
    * @param buffer Buffer to hold read data
    * @param nbytes Number of bytes to read
    * @param offset Offset in file to start reading from.
-   * @return Number of bytes actually read (< nbytes on error).
+   * @return Status
    */
-  static uint64_t read_all(
+  static Status read_all(
       int fd, void* buffer, uint64_t nbytes, uint64_t offset);
```

Comment:
not exploring callers, but thought we were discouraging return Status and using throw()s, 
maybe this change falls outside of that guidance due to other pre-existing code...

Reply:
It's temporary until we move to exceptions.